modules/Bio/EnsEMBL/Compara/RunnableDB/PrepareMaster/UpdateGenomesFromRegFactory.pm

```diff
@@ -15,39 +15,91 @@ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 See the License for the specific language governing permissions and
 limitations under the License.
 
-=cut
-
-=pod
-
 =head1 NAME
 
 Bio::EnsEMBL::Compara::RunnableDB::PrepareMaster::UpdateGenomesFromRegFactory
 
-=head1 SYNOPSIS
+=head1 DESCRIPTION
 
 Returns the list of species/genomes to add to the master database from the core
 databases in the registry file
 
+=head1 EXAMPLES
+
+    standaloneJob.pl Bio::EnsEMBL::Compara::RunnableDB::PrepareMaster::UpdateGenomesFromRegFactory \
+        -compara_db $(mysql-ens-compara-prod-9-ensadmin details url jalvarez_prep_vertebrates_master_for_rel_103) \
+        -master_db $(mysql-ens-compara-prod-1 details url ensembl_compara_master)
+
 =cut
 
 package Bio::EnsEMBL::Compara::RunnableDB::PrepareMaster::UpdateGenomesFromRegFactory;
 
 use warnings;
 use strict;
+
 use Bio::EnsEMBL::Registry;
+use Bio::EnsEMBL::Compara::Utils::CoreDBAdaptor;
 
 use base ('Bio::EnsEMBL::Hive::Process');
 
+
 sub fetch_input {
     my $self = shift;
+    # Get the core database adaptors for all the current species specified in the registry
     my $species_list = Bio::EnsEMBL::Registry->get_all_species();
-    $self->param('species_to_update', $species_list);
+    my %core_dbas;
+    foreach my $species_name ( @$species_list ) {
+        my $dba = Bio::EnsEMBL::Registry->get_DBAdaptor($species_name, 'core');
+        $core_dbas{$species_name} = $dba;
+    }
+
+    # If provided, get the list of allowed species
+    my $allowed_species_file = $self->param('allowed_species_file');
+    if (defined $allowed_species_file && -e $allowed_species_file) {
```

Comment:
```suggestion
    if (defined $allowed_species_file && -s $allowed_species_file) {
```
In case the file exists but is empty?

Reply:
That file should never be empty, so if it exists, being empty should remove all species from the release (that's the expected behaviour). "Unfortunately", with the current implementation `decode_json()` will fail because it doesn't accept empty strings. I think either I fix it to work as I have said before, or to die with an appropriate message if the file exits and it is empty (I prefer this one). I'm not keen on just ignoring the file because it is empty. Opinions?